lib/fhir/resources/healthcare-service.js

```diff
@@ -0,0 +1,168 @@
+    name: 'HealthcareService',
+
+    searchFilters: (ctx, callback) => {
+      const queryParams = ctx.query
+      const supportedParams = {
+        _id: {},
+        _lastUpdated: {},
+        active: {},
+        identifier: { allowArray: true },
+        location: {},
+        name: { allowArray: true, modifiers: { exact: true, contains: true } },
+        organization: {},
+        type: {},
+        'organization.active': {},
+        'organization.identifier': { allowArray: true },
+        'organization.name': { allowArray: true, modifiers: { exact: true, contains: true } },
+        'location.status': {},
+        'location.identifier': { allowArray: true },
+        'location.name': { allowArray: true, modifiers: { exact: true, contains: true } },
+        _format: {},
+        _include: { allowArray: true },
+        _revinclude: { allowArray: true }
+      }
+
+      queryUtils.validateAndParseQueryParams(queryParams, supportedParams, (badRequest, queryObject) => {
+        if (badRequest) {
+          return callback(null, fhirCommon.buildHTTPOutcome(400, 'error', 'invalid', badRequest))
+        }
+
+        let query = { $and: [] }
+        let promises = []
+
+        if (queryObject['_id']) {
+          query['$and'].push(queryUtils.tokenToMongoClause('id', queryObject['_id'][constants.NO_MODIFER]))
+        }
+
+        if (queryObject['_lastUpdated']) {
+          query['$and'].push(queryUtils.dateToMongoClause('meta.lastUpdated', queryObject['_lastUpdated'][constants.NO_MODIFER]))
+        }
+
+        if (queryObject['active']) {
+          query['$and'].push(queryUtils.boolToMongoClause('active', queryObject['active'][constants.NO_MODIFER]))
+        }
+
+        if (queryObject['identifier']) {
+          query['$and'].push(queryUtils.tokenToSystemValueElemMatch('identifier', queryObject['identifier'][constants.NO_MODIFER]))
+        }
+
+        if (queryObject['location']) {
+          query['$and'].push({ 'location.reference': queryUtils.paramAsReference(queryObject['location'][constants.NO_MODIFER], 'Location') })
+        }
+
+        if (queryObject['name']) {
+          for (const modifier in queryObject['name']) {
+            query['$and'].push(queryUtils.stringToMongoClause('name', queryObject['name'][modifier], modifier))
+          }
+        }
+
+        if (queryObject['organization']) {
+          query['$and'].push({ 'providedBy.reference': queryUtils.paramAsReference(queryObject['organization'][constants.NO_MODIFER], 'Organization') })
+        }
+
+        if (queryObject['type']) {
+          query['$and'].push(queryUtils.tokenToSystemCodeElemMatch('type.coding', queryObject['type'][constants.NO_MODIFER]))
+        }
+
+        if (queryObject['organization.active']) {
+          promises.push(new Promise((resolve, reject) => {
+            queryUtils.genChainedParamQuery(queryObject['organization.active'][constants.NO_MODIFER], 'organization.reference', 'active', organization, (err, clause) => {
```

Comment:
We have changed the way `genChainedParamQuery` works. It not accepts an array of fhirModules in the latest commits. We believe we will need to change how we are passing the fhir module to this function in each case where it's used in this PR. It shouldn't be too much different.